packages/medusa/src/models/payment-collection.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  BeforeInsert,
+  Column,
+  Index,
+  JoinColumn,
+  JoinTable,
+  ManyToMany,
+  ManyToOne,
+} from "typeorm"
+
+import { SoftDeletableEntity } from "../interfaces/models/soft-deletable-entity"
+import { DbAwareColumn } from "../utils/db-aware-column"
+import { generateEntityId } from "../utils"
+import { Currency, Payment, PaymentSession, Region } from "."
+
+import OrderEditingFeatureFlag from "../loaders/feature-flags/order-editing"
+import { FeatureFlagEntity } from "../utils/feature-flag-decorators"
+
+export enum PaymentCollectionStatus {
+  NOT_PAID = "not_paid",
+  AWAITING = "awaiting",
+  AUTHORIZED = "authorized",
+  PARTIALLY_AUTHORIZED = "partially_authorized",
+  CAPTURED = "captured",
+  PARTIALLY_CAPTURED = "partially_captured",
+  REFUNDED = "refunded",
+  PARTIALLY_REFUNDED = "partially_refunded",
+  CANCELED = "canceled",
+  REQUIRES_ACTION = "requires_action",
+}
+
+export enum PaymentCollectionType {
+  ORDER_EDIT = "order_edit",
+  SINGLE_PAYMENT = "single_payment",
+}
+
+@FeatureFlagEntity(OrderEditingFeatureFlag.key)
+export class PaymentCollection extends SoftDeletableEntity {
+  @DbAwareColumn({ type: "enum", enum: PaymentCollectionType })
+  type: string
+
+  @DbAwareColumn({ type: "enum", enum: PaymentCollectionStatus })
+  status: string
+
+  @Column({ nullable: true })
+  description: string
+
+  @Column({ type: "int" })
+  amount: number
+
+  @Column({ type: "int", nullable: true })
+  authorized_amount: number
+
+  @Column({ type: "int", nullable: true })
+  refunded_amount: number
+
+  @Index()
+  @Column()
+  region_id: string
+
+  @ManyToOne(() => Region)
+  @JoinColumn({ name: "region_id" })
+  region: Region
+
+  @Index()
+  @Column()
+  currency_code: string
+
+  @ManyToOne(() => Currency)
+  @JoinColumn({ name: "currency_code", referencedColumnName: "code" })
+  currency: Currency
+
+  @ManyToMany(() => PaymentSession)
+  @JoinTable({
+    name: "payment_collection_sessions",
+    joinColumn: {
+      name: "payment_collection_id",
+      referencedColumnName: "id",
+    },
+    inverseJoinColumn: {
+      name: "payment_session_id",
+      referencedColumnName: "id",
+    },
+  })
+  payment_sessions: PaymentSession[]
+
+  @ManyToMany(() => Payment)
+  @JoinTable({
+    name: "payment_collection_payments",
+    joinColumn: {
+      name: "payment_collection_id",
+      referencedColumnName: "id",
+    },
+    inverseJoinColumn: {
+      name: "payment_id",
+      referencedColumnName: "id",
+    },
+  })
+  payments: Payment[]
+
+  @DbAwareColumn({ type: "jsonb" })
+  metadata: Record<string, unknown>
+
+  @Column({ nullable: true })
+  created_by: string
```

Comment:
thought: Would it make sense to make this non-nullable? Do you have any scenarios in mind where `created_by` would not need to be populated? I would assume that even third-party consumers should be enforced to pass this information upon creating a payment collection

Reply:
I don't have a strong argument for that. Just thought that being something generic, it could have use cases where that wasn't required. 
I will remove it.